### ISSUE TITLE
Handlers combinations

### ISSUE THREAD

**Issue by jlmadurga** [python-telegram-bot/python-telegram-bot#166]
I have following scenario that I can not fullfil:
-   Bot can receive commands. `TelegramCommandHandler` and `TelegramUnknownCommandHandler`
-  Bot can receive messages to answer some command actions: `TelegramRegexHandler`
-  Bot can receive unknown messages, answering with help message or resending to human: `TelegramMessageHandler`

`TelegramRegexHandler` and `TelegramMessageHandler` are used for the same `Update`. This is the piece where both are dispatched cause they are in different block control:

```
        # Telegram update (regex)
        if isinstance(update, Update):
            self.dispatchRegex(update)
            handled = True

        # Telegram update (command)
        if isinstance(update, Update) \
                and update.message.text.startswith('/'):
            self.dispatchTelegramCommand(update)
            handled = True

        # Telegram update (message)
        elif isinstance(update, Update):
            self.dispatchTelegramMessage(update)
            handled = True
```

Is there any reason for this behavior? I don't think handling twice one request is a good option

**Comment by jh0ker** [comment]
As discussed with Yuri in the Telegram group, we should also add some kind of priority to regex handlers. Currently, all matching handlers are called.

**Comment by jlmadurga** [comment]
In my opinion dispatcher should call only to one handler giving priority in order of declaration. On the other hand, I will add handlers in more declarative way to see order in a glance
```
handlers = [command('start', callback_start),
            command('help', callback_help),
            regex(''^put_(?P<something>[-_a-zA-Z0-9]+)', callback_put_something),
           .....
           ]
```
Then, to add handlers and we can keep backwards compatibility only warning only one handler is executed and the order counts:

```
dispatcher.addHandlers(handlers)
```

**Comment by jh0ker** [comment]
@jlmadurga A very similar implementation is now online on https://github.com/python-telegram-bot/python-telegram-bot/tree/dispatcher-rework/telegram/ext

**Comment by leandrotoledo** [comment]
Resolving, please the implementation is now released under 4.0rc1, please update your library.

```
$ pip install python-telegram-bot==4.0rc1
```
